src/utils/cli.ts

```diff
@@ -60,76 +50,101 @@ export class CLI {
      */
     private parseCmdOptions() {
 
+        this.flags = {
+            process: this.commander.Process,
+            monitor: this.commander.Monitor,
+            add: this.commander.Add
+        };
+
         if (this.commander.dbLocation) {
-            if (!this.commander.dbLocation.split) {
+            let split = this.commander.dbLocation.split(":", 2);
+            if (split.length < 2) {
                 console.warn("invalid --db-location [host]:[port]");
             } else {
-                let splitted = this.commander.dbLocation.split(":", 2);
-                this.parameters.dbHost = splitted[0];
-                let dbPort = parseInt(splitted[1]);
-                if (dbPort) this.parameters.dbPort = dbPort;
+                this.parameters["dbHost"] = split[0];
+                this.parameters["dbPort"] = split[1];
             }
         }
 
         if (this.commander.dbAccess) {
-            if (!this.commander.dbAccess.split) {
+            let split = this.commander.dbAccess.split(":", 2);
+            if (split.length < 2) {
                 console.warn("invalid --db-access [user]:[password]");
             } else {
-                let splitted = this.commander.dbAccess.split(":", 2);
-                this.parameters.dbUser = splitted[0];
-                this.parameters.dbPW = splitted[1];
+                this.parameters["dbUser"] = split[0];
+                this.parameters["dbPW"] = split[1];
             }
         }
 
         if (this.commander.dbName) {
-            this.parameters.dbName = this.commander.dbName;
+            this.parameters["dbName"] = this.commander.dbName;
         }
 
         if (this.commander.blobLocation) {
-            if (!this.commander.blobLocation.split) {
+            let split = this.commander.blobLocation.split(":", 2);
+            if (split.length < 2) {
                 console.warn("invalid --blob-location [account]:[container]");
             } else {
-                let splitted = this.commander.blobLocation.split(":", 2);
-                this.parameters.blobAccount = splitted[0];
-                this.parameters.blobContainer = splitted[1];
+                this.parameters["blobAccount"] = split[0];
```

Comment:
`.blobAccount` requires you to define the object structure in TS, `["blobAccount"]` does not. 
Until now we had this big object with all its properties set to undefined. I had to change this as `ParamLoader` will fail to merge the parameters when they are defined, but set to undefined.